From a95dd50433cfa726eeeab73da3e27ed280b3236b Mon Sep 17 00:00:00 2001
From: Jake Mannens <jakem_5@hotmail.com>
Date: Wed, 12 Feb 2020 09:34:59 +1100
Subject: [PATCH 1/3] Separated text color into it's own Xresource.

---
 sxiv.1   | 5 ++++-
 sxiv.h   | 1 +
 window.c | 8 +++++---
 3 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/sxiv.1 b/sxiv.1
index eff3d66..81966cd 100644
--- a/sxiv.1
+++ b/sxiv.1
@@ -379,11 +379,14 @@ Zoom out.
 The following X resources are supported:
 .TP
 .B background
-Color of the window background and bar foreground
+Color of the window background
 .TP
 .B foreground
 Color of the window foreground and bar background
 .TP
+.B textColor
+Color of the text in the bar
+.TP
 .B font
 Name of Xft bar font
 .TP
diff --git a/sxiv.h b/sxiv.h
index 707eba7..359c454 100644
--- a/sxiv.h
+++ b/sxiv.h
@@ -409,6 +409,7 @@ struct win {
 
 	XftColor bg;
 	XftColor fg;
+	XftColor tc;
 
 	int x;
 	int y;
diff --git a/window.c b/window.c
index 6f9a390..361fcdf 100644
--- a/window.c
+++ b/window.c
@@ -92,7 +92,7 @@ const char* win_res(XrmDatabase db, const char *name, const char *def)
 void win_init(win_t *win)
 {
 	win_env_t *e;
-	const char *bg, *fg, *f;
+	const char *bg, *fg, *tc, *f;
 	char *res_man;
 	XrmDatabase db;
 
@@ -121,8 +121,10 @@ void win_init(win_t *win)
 
 	bg = win_res(db, RES_CLASS ".background", "white");
 	fg = win_res(db, RES_CLASS ".foreground", "black");
+        tc = win_res(db, RES_CLASS ".textColor", "white");
 	win_alloc_color(e, bg, &win->bg);
 	win_alloc_color(e, fg, &win->fg);
+        win_alloc_color(e, tc, &win->tc);
 
 	win->bar.l.size = BAR_L_LEN;
 	win->bar.r.size = BAR_R_LEN;
@@ -408,12 +410,12 @@ void win_draw_bar(win_t *win)
 			return;
 		x = win->w - tw - H_TEXT_PAD;
 		w -= tw;
-		win_draw_text(win, d, &win->bg, x, y, r->buf, len, tw);
+		win_draw_text(win, d, &win->tc, x, y, r->buf, len, tw);
 	}
 	if ((len = strlen(l->buf)) > 0) {
 		x = H_TEXT_PAD;
 		w -= 2 * H_TEXT_PAD; /* gap between left and right parts */
-		win_draw_text(win, d, &win->bg, x, y, l->buf, len, w);
+		win_draw_text(win, d, &win->tc, x, y, l->buf, len, w);
 	}
 	XftDrawDestroy(d);
 }

From c69d423cf49ec7adbb81d4b5d38a48b48562e00f Mon Sep 17 00:00:00 2001
From: Jake Mannens <jakem_5@hotmail.com>
Date: Wed, 12 Feb 2020 17:27:12 +1100
Subject: [PATCH 2/3] Changed textColor to barBackground.

---
 sxiv.1   |  8 ++++----
 sxiv.h   |  2 +-
 window.c | 10 +++++-----
 3 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/sxiv.1 b/sxiv.1
index 81966cd..a2a04df 100644
--- a/sxiv.1
+++ b/sxiv.1
@@ -379,13 +379,13 @@ Zoom out.
 The following X resources are supported:
 .TP
 .B background
-Color of the window background
+Color of the window background and bar foreground
 .TP
 .B foreground
-Color of the window foreground and bar background
+Color of the window foreground
 .TP
-.B textColor
-Color of the text in the bar
+.B barBackground
+Color of the bar background
 .TP
 .B font
 Name of Xft bar font
diff --git a/sxiv.h b/sxiv.h
index 359c454..d4de430 100644
--- a/sxiv.h
+++ b/sxiv.h
@@ -409,7 +409,7 @@ struct win {
 
 	XftColor bg;
 	XftColor fg;
-	XftColor tc;
+	XftColor bbg;
 
 	int x;
 	int y;
diff --git a/window.c b/window.c
index 361fcdf..431a705 100644
--- a/window.c
+++ b/window.c
@@ -92,7 +92,7 @@ const char* win_res(XrmDatabase db, const char *name, const char *def)
 void win_init(win_t *win)
 {
 	win_env_t *e;
-	const char *bg, *fg, *tc, *f;
+	const char *bg, *fg, *bbg, *f;
 	char *res_man;
 	XrmDatabase db;
 
@@ -121,10 +121,10 @@ void win_init(win_t *win)
 
 	bg = win_res(db, RES_CLASS ".background", "white");
 	fg = win_res(db, RES_CLASS ".foreground", "black");
-        tc = win_res(db, RES_CLASS ".textColor", "white");
+        bbg = win_res(db, RES_CLASS ".barBackground", "white");
 	win_alloc_color(e, bg, &win->bg);
 	win_alloc_color(e, fg, &win->fg);
-        win_alloc_color(e, tc, &win->tc);
+        win_alloc_color(e, bbg, &win->bbg);
 
 	win->bar.l.size = BAR_L_LEN;
 	win->bar.r.size = BAR_R_LEN;
@@ -410,12 +410,12 @@ void win_draw_bar(win_t *win)
 			return;
 		x = win->w - tw - H_TEXT_PAD;
 		w -= tw;
-		win_draw_text(win, d, &win->tc, x, y, r->buf, len, tw);
+		win_draw_text(win, d, &win->bbg, x, y, r->buf, len, tw);
 	}
 	if ((len = strlen(l->buf)) > 0) {
 		x = H_TEXT_PAD;
 		w -= 2 * H_TEXT_PAD; /* gap between left and right parts */
-		win_draw_text(win, d, &win->tc, x, y, l->buf, len, w);
+		win_draw_text(win, d, &win->bbg, x, y, l->buf, len, w);
 	}
 	XftDrawDestroy(d);
 }

From 3ac6e6c56fd22366c05d0fb75b1296da210b0779 Mon Sep 17 00:00:00 2001
From: Jake Mannens <jakem_5@hotmail.com>
Date: Wed, 12 Feb 2020 17:42:16 +1100
Subject: [PATCH 3/3] Forgot to actually swap bar colors.

---
 window.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/window.c b/window.c
index 431a705..2d5e0a9 100644
--- a/window.c
+++ b/window.c
@@ -399,7 +399,7 @@ void win_draw_bar(win_t *win)
 	d = XftDrawCreate(e->dpy, win->buf.pm, DefaultVisual(e->dpy, e->scr),
 	                  DefaultColormap(e->dpy, e->scr));
 
-	XSetForeground(e->dpy, gc, win->fg.pixel);
+	XSetForeground(e->dpy, gc, win->bbg.pixel);
 	XFillRectangle(e->dpy, win->buf.pm, gc, 0, win->h, win->w, win->bar.h);
 
 	XSetForeground(e->dpy, gc, win->bg.pixel);
@@ -410,12 +410,12 @@ void win_draw_bar(win_t *win)
 			return;
 		x = win->w - tw - H_TEXT_PAD;
 		w -= tw;
-		win_draw_text(win, d, &win->bbg, x, y, r->buf, len, tw);
+		win_draw_text(win, d, &win->fg, x, y, r->buf, len, tw);
 	}
 	if ((len = strlen(l->buf)) > 0) {
 		x = H_TEXT_PAD;
 		w -= 2 * H_TEXT_PAD; /* gap between left and right parts */
-		win_draw_text(win, d, &win->bbg, x, y, l->buf, len, w);
+		win_draw_text(win, d, &win->fg, x, y, l->buf, len, w);
 	}
 	XftDrawDestroy(d);
 }
